src/analyzer/pil_analyzer.rs

```diff
@@ -260,7 +260,7 @@ impl PILContext {
     }
 
     fn handle_namespace(&mut self, name: &str, degree: &ast::Expression) {
-        self.polynomial_degree = abstract_to_degree(&self.evaluate_expression(degree).unwrap());
```

Comment:
Can you maybe add a comment that this should in the end be done in arbitrary precision instead?

Reply:
Do you mean as opposed to on the field element value?

Reply:
yes